### PR TITLE
[dpi,aes] Allow partial last message blocks for AES-GCM

### PR DESCRIPTION
**Pull request by andrea-caforio**
Unlike the existing AES modes, the new GCM mode allows for partial last message blocks.
This commit adds this option to the `c_dpi` implementation, i.e., the automatic padding of
partial blocks.

### CODE REVIEW

File: hw/ip/aes/dv/aes_model_dpi/aes_model_dpi.c [hunk 1]
```diff
@@ -210,28 +224,20 @@ void c_dpi_aes_crypt_message(unsigned char impl_i, unsigned char op_i,
...
-
   if (impl == 0) {
     // The C model is currently not supported.
     printf(
         "ERROR: c_dpi_aes_crypt_message() currently supports OpenSSL/BoringSSL "
         "only\n");
   } else {  // OpenSSL/BoringSSL
     if (!op) {
-      crypto_encrypt(ref_out, iv, ref_in, data_len, key, key_len, mode, aad_in,
+      crypto_encrypt(ref_out, iv, ref_in, data_len, key, key_len, kCryptoAesGcm, aad_in,
```
Comment (nasahlpa) [inline hw/ip/aes/dv/aes_model_dpi/aes_model_dpi.c]: Is this change intentional?

**Review by vogelpi — APPROVED** [review]
This looks good to me except for the point raised by @nasahlpa . Thanks for the PR @andrea-caforio !

**Comment by vogelpi** [comment]
Also, it currently fails lint. Can you please do a `clang-format -i` on this?

File: hw/ip/aes/dv/aes_model_dpi/aes_model_dpi.c (hunk 1 above)
Comment (andrea-caforio) [reply hw/ip/aes/dv/aes_model_dpi/aes_model_dpi.c]: Nice catch. Thank you. :-)

**Comment by andrea-caforio** [comment]
It should be alright now (except for the lint error in that unrelated DV file).
Thanks for the reviews @vogelpi @nasahlpa.

**Review by nasahlpa — APPROVED** [review]
Thanks for working on that!
